src/main.rs

```diff
@@ -0,0 +1,66 @@
+use rand::Rng;
+use std::fmt;
+
+fn main() {
+    let mut bag = TileBag::default();
+    for i in 0..1000 {
+        let tile = bag.draw_tile();
+        println!("{}", tile)
+    }
+}
+
+struct TileBag {
+    bag: Vec<char>,
+    rng: rand::rngs::ThreadRng,
+    letter_distribution: [usize; 26],
+}
+
+impl TileBag {
+    fn new(letter_distribution: [usize; 26]) -> Self {
+        let mut tile_bag = TileBag {
+            bag: Vec::new(),
+            rng: rand::thread_rng(),
+            letter_distribution,
+        };
+        tile_bag.fill();
+        tile_bag
+    }
+
+    fn draw_tile(&mut self) -> char {
+        if (self.bag.len() == 0) {
```

Comment:
```suggestion
        if self.bag.is_empty() {
```

If you don't have clippy lints turned on in your IDE I would do so, I believe this would be picked up there. `is_empty` is favored as it signals intent better, and I too have a bad habit of wrapping my if conditions in parentheses 😅